lib/active_merchant/billing/gateways/securion_pay.rb

```diff
@@ -133,13 +133,48 @@ def create_post_for_auth_or_purchase(money, payment, options)
         add_creditcard(post, payment, options)
         add_customer(post, payment, options)
         add_customer_data(post, options)
+        add_three_ds(post, options)
         if options[:email]
           post[:metadata] = {}
           post[:metadata][:email] = options[:email]
         end
         post
       end
 
+      def add_three_ds(post, options)
+        return if options[:three_ds_options].blank?
+
+        post[:threeDSecure] = {
+          external: {
+            version: options[:three_ds_options][:version],
+            authenticationValue: options[:three_ds_options][:cavv],
+            acsTransactionId: options[:three_ds_options][:acs_transaction_id],
+            status: options[:three_ds_options][:authentication_response_status],
+            eci: options[:three_ds_options][:ecommerce_indicator]
+          }.merge(xid_or_ds_trans_id(options[:three_ds_options]))
+        }
+      end
+
+      def xid_or_ds_trans_id(three_ds)
+        if three_ds[:version].to_f >= 2.0
+          { dsTransactionId: three_ds[:ds_transaction_id] }
+        else
+          { xid: three_ds[:xid] }
+        end
+      end
+
+      def validate_three_ds_params(three_ds)
+        errors = {}
+        supported_version = %w{1.0.2 2.1.0 2.2.0}.include?(three_ds[:version])
+        supported_auth_response = ['Y', 'N', 'U', 'R', 'E', 'A', nil].include?(three_ds[:status])
+
+        errors[:three_ds_version] = 'ThreeDs version not supported' unless supported_version
+        errors[:auth_response] = 'Authentication response value not supported' unless supported_auth_response
+        errors.compact!
+
+        errors.present? ? Response.new(false, 'ThreeDs data is invalid', errors) : nil
+      end
```

Comment:
Lets remove internal validation for now, pending discussion in Slack.

Reply:
After discussion, let's proceed with the validation, as it provides value for testing